sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2372,10 +2375,30 @@ class AstBuilder extends DataTypeAstBuilder
 
         val tvfAliases = if (aliases.nonEmpty) UnresolvedTVFAliases(ident, tvf, aliases) else tvf
 
-        tvfAliases.optionalMap(func.tableAlias.strictIdentifier)(aliasPlan)
+        tvfAliases.optionalMap(ctx.tableAlias.strictIdentifier)(aliasPlan)
       })
   }
 
+  override def visitStreamTableValuedFunction(ctx: StreamTableValuedFunctionContext): LogicalPlan =
+    withOrigin(ctx) {
+      visitFunctionTable(ctx.functionTable).transformUp {
+        case tvf: UnresolvedTableValuedFunction => tvf.copy(isStreaming = true)
```

Comment:
let's pass a parameter `isStreaming` to `visitFunctionTable` and make it a private method instead of overriding it.